xdsl/dialects/riscv.py

```diff
@@ -144,13 +144,18 @@ def register_name(self) -> str:
             raise ValueError("Cannot get name for unallocated register")
         return self.data.name
 
+    @property
+    def is_allocated(self) -> bool:
+        """Returns true if a RISCV register is allocated, otherwise false"""
+        return self.data.name is not None
```

Comment:
```suggestion
        return self.data.is_allocated
```
Could you please add this as a helper on the register directly?